android/src/main/java/com/reactnative/googlecast/GoogleCastModule.java

```diff
@@ -14,288 +16,352 @@
+                        .setStreamType(MediaInfo.STREAM_TYPE_BUFFERED)
+                        .setMetadata(movieMetadata);
 
-    if (params.hasKey("subtitle") && params.getString("subtitle") != null) {
-      movieMetadata.putString(MediaMetadata.KEY_SUBTITLE,
-                              params.getString("subtitle"));
-    }
+        if (params.hasKey("contentType") &&
+                params.getString("contentType") != null) {
+            builder = builder.setContentType(params.getString("contentType"));
+        } else {
+            builder = builder.setContentType("video/mp4");
+        }
 
-    if (params.hasKey("studio") && params.getString("studio") != null) {
-      movieMetadata.putString(MediaMetadata.KEY_STUDIO,
-                              params.getString("studio"));
-    }
+        if (params.hasKey("streamDuration")) {
+            builder = builder.setStreamDuration(params.getInt("streamDuration"));
+        }
 
-    if (params.hasKey("imageUrl") && params.getString("imageUrl") != null) {
-      movieMetadata.addImage(
-          new WebImage(Uri.parse(params.getString("imageUrl"))));
+        return builder.build();
     }
 
-    if (params.hasKey("posterUrl") && params.getString("posterUrl") != null) {
-      movieMetadata.addImage(
-          new WebImage(Uri.parse(params.getString("posterUrl"))));
+    @ReactMethod
+    public void getCastState(final Promise promise) {
+        getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+            @Override
+            public void run() {
+                CastContext castContext =
+                        CastContext.getSharedInstance(getReactApplicationContext());
+                promise.resolve(castContext.getCastState() - 1);
+            }
+        });
     }
 
-    MediaInfo.Builder builder =
-        new MediaInfo.Builder(params.getString("mediaUrl"))
-            .setStreamType(MediaInfo.STREAM_TYPE_BUFFERED)
-            .setMetadata(movieMetadata);
 
-    if (params.hasKey("contentType") &&
-        params.getString("contentType") != null) {
-      builder = builder.setContentType(params.getString("contentType"));
-    } else {
-      builder = builder.setContentType("video/mp4");
+    @ReactMethod
+    public void initChannel(final String namespace) {
+        if (mCastSession != null) {
+            getReactApplicationContext().runOnUiQueueThread(new Runnable() {
+                @Override
+                public void run() {
+                    try {
+                        mCastSession.setMessageReceivedCallbacks(namespace, new Cast.MessageReceivedCallback() {
+                            @Override
+                            public void onMessageReceived(CastDevice castDevice, String channelNameSpace, String message) {
+                                WritableMap map = Arguments.createMap();
+                                map.putString("getFriendlyName", castDevice.getFriendlyName());
```

Comment:
This doesn't have iOS equivalent. Maybe just call it `deviceName` for now, it's being completely reworked in v4 anyway...